app/src/main/java/com/example/myapplication/CredentialsManager.kt

```diff
@@ -9,16 +10,34 @@ class CredentialsManager {
             "[a-zA-Z0-9][0-zA-Z0-9\\-]{0,25}" +
             ")+").toRegex()
 
+    private val registeredUsers = mutableMapOf<String, String>()
+
     fun isEmailValid(email: String): Boolean {
         return email.matches(emailPattern)
     }
 
+    fun isEmailAlreadyUsed(email: String): Boolean {
+        return registeredUsers.containsKey(email.lowercase())
+    }
+
     fun isValidPassword(password: String): Boolean {
         return password.length >= 8
     }
 
-    fun validateCredentials(email: String, password: String, isCheckboxChecked: Boolean): Boolean {
-        return isEmailValid(email) && isValidPassword(password) && isCheckboxChecked
+    fun registerUser(email: String, password: String): Boolean {
+        if (isEmailAlreadyUsed(email)) {
+            return false
+        }
+        registeredUsers[email.lowercase()] = password
+        return true
+    }
+
+    fun validateLogin(email: String, password: String): Boolean {
```

Comment:
I don't think you validate login here - but just login instead
I know we don't set the flag if the user is logged in yet, but generally the goal is to log in the user, not to see if he can log in